firebaseai/LiveAudioExample/ViewModels/LiveViewModel.swift

```diff
@@ -0,0 +1,295 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseAI
+import Foundation
+import OSLog
+import AVFoundation
+import SwiftUI
+import AVKit
+
+enum LiveViewModelState {
+  case idle
+  case connecting
+  case connected
+}
+
+@MainActor
+class LiveViewModel: ObservableObject {
+  private var logger = Logger(subsystem: Bundle.main.bundleIdentifier!, category: "generative-ai")
+
+  @Published
+  var error: Error?
+
+  @Published
+  var state: LiveViewModelState = .idle
+
+  @Published
+  var transcriptViewModel: TranscriptViewModel = TranscriptViewModel()
+
+  @Published
+  var backgroundColor: Color? = nil
+
+  private var model: LiveGenerativeModel?
+  private var liveSession: LiveSession?
+
+  private var audioController: AudioController?
+  private var microphoneTask = Task<Void, Never> {}
+
+  init(firebaseService: FirebaseAI, backend: BackendOption) {
+    model = firebaseService.liveModel(
+      modelName: (backend == .googleAI) ? "gemini-live-2.5-flash-preview" : "gemini-2.0-flash-exp",
+      generationConfig: LiveGenerationConfig(
+        responseModalities: [.audio],
+        speech: SpeechConfig(voiceName: "Zephyr", languageCode: "en-US"),
+        outputAudioTranscription: AudioTranscriptionConfig()
+      ),
+      tools: [
+        .functionDeclarations([
+          FunctionDeclaration(
+            name: "changeBackgroundColor",
```

Comment:
It would be good to add an instruction text to the view to let users know they can say "change the color to blue".

Love the feature, it's fun!